bin/utils/node-utils.sh

```diff
@@ -0,0 +1,144 @@
+#!/bin/sh
+
+################################################################################
+# This program and the accompanying materials are made available under the terms of the
+# Eclipse Public License v2.0 which accompanies this distribution, and is available at
+# https://www.eclipse.org/legal/epl-v20.html
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Copyright IBM Corporation 2020
+################################################################################
+
+# TODO LATER - anyway to do this better?
+# Try and work out where we are even if sourced
+if [[ -n ${INSTALL_DIR} ]]
+then
+  export utils_dir="${INSTALL_DIR}/bin/utils"
+elif [[ -n ${ZOWE_ROOT_DIR} ]]
+then
+  export utils_dir="${ZOWE_ROOT_DIR}/bin/utils"
+elif [[ -n ${ROOT_DIR} ]]
+then
+  export utils_dir="${ROOT_DIR}/bin/utils"
+elif [[ $0 == "node-utils.sh" ]] #Not called by source
+then
+  export utils_dir=$(cd $(dirname $0);pwd)
+else
+  echo "Could not work out the path to the utils directory. Please 'export ZOWE_ROOT_DIR=<zowe-install-directory' before running." 1>&2
+  exit 1
+fi
+
+# Source common util functions
+. ${utils_dir}/common.sh
+
+# TODO - how to test well given interaction and guess?
+# Interactive function that checks if the current NODE_HOME is valid and if not requests a user enters the node home path via command line
+prompt_node_home() {
```

Comment:
I guess there are a couple of problems:
1. These scripts are called a fair way down the stack, so passing a flag through all is not great. We could make a opt on zowe-configure-instance.sh/zowe-setup-certificates.sh to make it non interactive and export a shell variable based on that? I'm not sure if `-i` would be confusing for non-interactive though!
2. What behaviour do we want for non-interactive? We obviously don't want to exit the script, so just print an error and continue?

Reply:
`NODE_HOME` is mandatory in this case and for future steps to start Zowe, so I think if it's non-interactive mode, it's safe we prompt the error and exit the script.

Reply:
My concern is that not all of our scripts are idempotent, so I don't know if we will hit issues when the user fixes the issue then re-runs. Do you mind if I create a new issue for this rather than have to solve it right now?

Reply:
Yes that sounds good to me. Thanks!

Reply:
Captured in https://github.com/zowe/zowe-install-packaging/issues/1374